Sources/DefaultsController.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+enum UserDefaultsState: Equatable {
+    case standard
+    case appGroup(String)
+
+    static func makeState(appGroupID: String?) -> UserDefaultsState {
+        if let appGroupID = appGroupID {
+            return .appGroup(appGroupID)
+        } else {
+            return .standard
+        }
+    }
+
+    static func == (lhs: UserDefaultsState, rhs: UserDefaultsState) -> Bool {
+        switch (lhs, rhs) {
+        case (.standard, .standard):
+            return true
+        case (.appGroup(let group1), .appGroup(let group2)):
+            return group1 == group2
+        default:
+            return false
+        }
+    }
+}
+
+struct DefaultsController {
```

Comment:
`DefaultsController` is a bit ambiguous to me, can we call this `TimeStorage` or the like? What do you think?

Reply:
Also why having this as `static` functions and global state vs an actual instance? I'd expect consumers to do `self.storage = TimeStorage()` if that makes sense?

Reply:
Yeah that’s a good call